Print object enhancements 5

slyrus opened the pull request:
These commits allow for printing some CLIM objects readably when *print-readably* is t.

nsrahmad commented:
@slyrus I've seen `*print-readably*` used for serialization, but never for debugging. Do you mind telling how you used it?

slyrus commented:
@nsrahmad as an example of where I would like to use this, I'm currently trying to track down a problem with arc drawing in the PDF backend. I see this frame the output of a trace of my transform-angle function: #<CLIM-INTERNALS::STANDARD-HAIRY-TRANSFORMATION 1 0 0 -1 0 842>. It would be great if I were to set *print-readably* to t and have the output be something like #.(make-transformation 1 0 0 -1 0 842). Yes, I can do this by hand, but why bother? Particularly for the immutable classes, this seems like a win.

scymtym commented on Core/clim-basic/regions.lisp:
```diff
@@ -551,12 +551,13 @@
+  (print-unreadable-object (ell stream :type t :identity t)
+    (with-slots (start-angle end-angle tr) ell
+      (format stream "~A [~A ~A] ~A"
+              (type-of ell)
```
I don't know if this has been mentioned before, but `print-unreadable-object … :type t` already prints the type so this shouldn't be needed.

slyrus commented:
I have addressed the requested changes and rebased to the new master branch. Closing this PR and opening a new one. Not sure if there's a better way to deal with conflicts/rebasing than that, but here goes.
